### ISSUE TITLE
Crash with Mapbox-iOS-SDK 2.0.0-pre.1

### ISSUE THREAD

**Issue by webventil** [mapbox/mapbox-gl-native#2155]
I got a crash in the latest Mapbox-iOS-SDK version. Unfortunately I don't know how to reproduce. The app was in Background at that time logging gps coordinates.

What I released is that there where 19 threads that have the same stack trace.
```
Thread 5:
0   libsystem_kernel.dylib        	0x000000019821ce0c mach_msg_trap + 8
1   libsystem_kernel.dylib        	0x000000019821cc84 mach_msg + 68
2   CoreFoundation                	0x0000000185d5f46c __CFRunLoopServiceMachPort + 196
3   CoreFoundation                	0x0000000185d5d3c0 __CFRunLoopRun + 936
4   CoreFoundation                	0x0000000185c88f70 CFRunLoopRunSpecific + 392
5   CoreFoundation                	0x0000000185cdaff8 CFRunLoopRun + 108
6   Mapbox                        	0x00000001008faeec uv__cf_loop_runner (darwin.c:119)
7   Mapbox                        	0x00000001008f983c uv__thread_start (uv-common.c:322)
8   libsystem_pthread.dylib       	0x00000001982d3db4 _pthread_body + 160
9   libsystem_pthread.dylib       	0x00000001982d3d10 _pthread_start + 156
10  libsystem_pthread.dylib       	0x00000001982d0ee4 thread_start + 0
```

The crashed thread stack trace:
```
Thread 47 name:  Map
Thread 47 Crashed:
0   libGPUSupportMercury.dylib    	0x000000018ef9df90 gpus_ReturnNotPermittedKillClient + 12
1   libGPUSupportMercury.dylib    	0x000000018ef9ef3c gpusSubmitDataBuffers + 168
2   AGXGLDriver                   	0x00000001844c845c agxuFinishContextResource + 100
3   AGXGLDriver                   	0x00000001844bb9bc glrWriteTextureData + 912
4   libGPUSupportMercury.dylib    	0x000000018efa59d0 gldModifyTexSubImage + 416
5   GLEngine                      	0x0000000189a591c4 glTexSubImage2D_Exec + 1804
6   OpenGLES                      	0x0000000189b3ea94 glTexSubImage2D + 104
7   Mapbox                        	0x0000000100802460 mbgl::LineAtlas::bind() (line_atlas.cpp:169)
8   Mapbox                        	0x0000000100802328 mbgl::LineAtlas::addDash(std::__1::vector<float, std::__1::allocator<float> > const&, bool) (line_atlas.cpp:128)
9   Mapbox                        	0x0000000100801fd4 mbgl::LineAtlas::getDashPosition(std::__1::vector<float, std::__1::allocator<float> > const&, bool) (line_atlas.cpp:42)
10  Mapbox                        	0x0000000100845e58 mbgl::Painter::renderLine(mbgl::LineBucket&, mbgl::StyleLayer const&, mbgl::TileID const&, std::__1::array<float, 16ul> const&) (painter_line.cpp:68)
11  Mapbox                        	0x0000000100844360 void mbgl::Painter::renderPass<std::__1::__wrap_iter<mbgl::RenderItem const*> >(mbgl::RenderPass, std::__1::__wrap_iter<mbgl::RenderItem const*>, std::__1::__wrap_iter<mbgl::RenderItem const*>, unsigned long, signed char, float) (painter.cpp:294)
12  Mapbox                        	0x0000000100842db0 mbgl::Painter::render(mbgl::Style const&, mbgl::TransformState, mbgl::FrameData const&, std::__1::chrono::time_point<std::__1::chrono::steady_clock, std::__1::chrono::duration<long long, std::__1::ratio<1l, 1000000000l> > >) (painter.cpp:241)
13  Mapbox                        	0x0000000100821514 mbgl::MapContext::renderSync(mbgl::TransformState const&, mbgl::FrameData const&) (map_context.cpp:331)
14  Mapbox                        	0x000000010081e7bc std::__1::__packaged_task_func<std::__1::__bind<mbgl::MapContext::RenderResult (mbgl::MapContext::*&)(mbgl::TransformState const&, mbgl::FrameData const&), mbgl::MapContext*&, mbgl::TransformState const&, mbgl::FrameData&>, std::__1::allocator<std::__1::__bind<mbgl::MapContext::RenderResult (mbgl::MapContext::*&)(mbgl::TransformState const&, mbgl::FrameData const&), mbgl::MapContext*&, mbgl::TransformState const&, mbgl::FrameData&> >, mbgl::MapContext::RenderResult ()>::operator()() (future:1817)
15  Mapbox                        	0x000000010081e334 std::__1::packaged_task<mbgl::MapContext::RenderResult ()>::operator()() (future:2090)
16  Mapbox                        	0x000000010081e288 mbgl::util::RunLoop::Invoker<std::__1::packaged_task<mbgl::MapContext::RenderResult ()>, std::__1::tuple<> >::operator()() (run_loop.hpp:90)
17  Mapbox                        	0x00000001008a4b2c mbgl::util::RunLoop::process() (run_loop.cpp:27)
18  Mapbox                        	0x00000001008e2da8 uv__async_event (async.c:80)
19  Mapbox                        	0x00000001008e3160 uv__async_io (async.c:156)
20  Mapbox                        	0x00000001008fc76c uv__io_poll (kqueue.c:233)
21  Mapbox                        	0x00000001008e380c uv_run (core.c:317)
22  Mapbox                        	0x000000010081f918 void mbgl::util::Thread<mbgl::MapContext>::run<std::__1::tuple<mbgl::View&, mbgl::FileSource&, mbgl::MapData&>, 0ul, 1ul, 2ul>(mbgl::util::ThreadContext, std::__1::tuple<mbgl::View&, mbgl::FileSource&, mbgl::MapData&>&&, std::__1::integer_sequence<unsigned long, 0ul, 1ul, 2ul>) (thread.hpp:125)
23  Mapbox                        	0x000000010081f814 std::__1::__thread_proxy<std::__1::tuple<mbgl::util::Thread<mbgl::MapContext>::Thread<mbgl::View&, mbgl::FileSource&, mbgl::MapData&>(mbgl::util::ThreadContext const&, mbgl::View&&&, mbgl::FileSource&&&, mbgl::MapData&&&)::'lambda'()> >(void*, void*) (thread.hpp:104)
24  libsystem_pthread.dylib       	0x00000001982d3db4 _pthread_body + 160
25  libsystem_pthread.dylib       	0x00000001982d3d10 _pthread_start + 156
26  libsystem_pthread.dylib       	0x00000001982d0ee4 thread_start + 0
```

**Comment by 1ec5** [comment]
Dupe of #1460, fixed in #1869. v2.0.0-pre.1 is just a renumbered v0.5.4 with some naming changes. However, the final v2.0.0 will contain this fix among others.

**Comment by incanus** [comment]
> v2.0.0-pre.1 is just a renumbered v0.5.4

v0.5.1 actually, but yes.
